### ISSUE TITLE
Status of the project

### ISSUE THREAD

**Issue by haarp** [backuppc/backuppc#518]
Hello,

is BackupPC still maintained @craigbarratt ?

I'm asking because this repo has not seen a commit, issue or issue comment by the author in two years now. The issue count keeps rising, and there are a couple of open PRs with no progress towards merging.

There are some issues like #437 which can lead to surprises and problems, but would be easy to fix. Other issues like #252 show potential interoperability issues with some tools which used to work differently, showing BackupPC's aging progress.

BackupPC is a great tool that sees a lot of use by people and organizations, with more popping up all the time. It would be a shame if it started getting buried under the sands of time.

Cheers!

**Comment by LordSimal** [comment]
https://github.com/backuppc/backuppc/issues/481

Just want to link issues together here. I have no idea if this project is still maintained.

**Comment by michieltimmers** [comment]
I'm also very interested in the current status. Need to reinstall my backup server and i'm wondering if I still should use backuppc

**Comment by squash** [comment]
Hoping for some good news here. I know it can be a pain to maintain these projects over decades but I'm hoping not to have to start shopping around for an alternative!

**Comment by Benoit75** [comment]
Hello, 
Same question here !

**Comment by rgristroph** [comment]
I also would like to see this project continue, I don't know of a good replacement for it, and I was about to set up a new backuppc server and configure a bunch of clients, but now I'm worried about it.  I am not sure bacula or duplicati share similar features, particularly the web based interface to restore particular files.

Do we need to fork it and set up some kind of committee to keep it alive ?

**Comment by freelock** [comment]
Subscribing... we've been using this for decades now, was just checking for updates as part of our regular maintenance schedule...

I just had to grow the backup pool filesystem -- I'd love to see work towards supporting object storage.

**Comment by tonimore** [comment]
Also want to know is  the project  still alive?

**Comment by aqueos** [comment]
hi there.

I am interested as well, i also proposed to help fund maintenance and new feature if needed.

regards,
Ghislain.

**Comment by opoplawski** [comment]
It seems like @larstobi and @moisseev are at least still active on github.  Perhaps they could add more people to the project to keep it alive?  I'd like to be able to help out, but I'm afraid I don't have much time.  I at least have some Perl experience.

**Comment by apio-sys** [comment]
Would be great if the project could be kept alive. Much better then creating a fork. These packages are shipped in Debian by default which is really nice. You will be a few years out before replacing that with a fork. As @opoplawski  I'ld like to see this project continue and can make myself avail for some tasks to contribute as I do on other FOSS projects. Not much, but if all give a bit, we'll get a long way. The project is very stable and works perfectly in modern production. We just need a few fixes here and there and then maybe some features but that is - IMHO - not the most important.

**Comment by wltechblog** [comment]
Being a Perl application I'm content to use what I have until a good replacement arrives. I consider writing a functionally similar system in Go, but I don't see anyone being enthusiastic about working on a Perl codebase in 2024.

**Comment by moisseev** [comment]
> Perhaps they could add more people to the project to keep it alive?

Hi everyone!

Technically, I could add members to the project, but since this is @craigbarratt's project, I won’t do so unless he requests it.

**Comment by klaernie** [comment]
@moisseev could you maybe message @craigbarratt  outside of GitHub to weigh in on what his intentions are? After all it might just be not much time left in his calendar to actually do work on OSS, but he might also not be opposed to accepting a new generation of maintainers. I just would like to prevent us all flooding his channels and being annoying, and you already are a known person to him.

**Comment by moisseev** [comment]
I don’t have any contacts for @craigbarratt outside of GitHub, so I’m unable to reach out directly. I understand the interest in bringing in a new generation of maintainers, but is there actually a list of specific people ready to lead the project?

**Comment by klaernie** [comment]
I would suggest LinkedIn as an alternative contact.

I'd also apply for a reviewer and triage role, since I'm not knowledgeable in the codebase I think this is a fitting way to get to grok the infrastructure.

**Comment by NicolasGoeddel** [comment]
I already had a few issues with Ubuntu 24.04 and a newer Perl version. I had to fix a few lines to make it work again.
But since I am working on my new NAS right now I guess I will move from BackupPC to another solution although I liked it a lot since the last 10+ years. But without any maintenance since years now I don't want to use it anymore in a productive environment.
I am software developer myself but have absolutely no experience with Perl. Would be much nicer to see Python here.

Anyway... In case this project will be maintained again I might use it again too.

**Comment by nixcamic** [comment]
Yeah I was just thinking it would be good to maybe publicize that a maintainer is wanted. There are probably people using backupPC who have the skills who would be willing to help out if it was known that a maintainer was actively needed.

**Comment by wltechblog** [comment]
As great as this software is, I can't imagine many people looking for an old Perl 5 project to take on. Much more likely to find someone willing to do a replacement project in a modern language. If the original maintainer has truly lost interest, it's pretty safe to assume this project is functionally EOL.

With that said, I've been using it for many years and appreciate the work that went into it, maybe the person doing the rewrite will end up being me.

**Comment by wbraswell** [comment]
Hello Everyone,

I am the President of the Perl Community 501(c)(3) nonprofit organization, and was asked by a user of this software to come investigate the possibility of our group adopting the project.

[https://perlcommunity.org/](https://perlcommunity.org/)

Without having ever used BackupPC myself, I would make a very rough estimation that it will probably cost approximately $2K to $3K per year for our nonprofit to hire one of our permanent Perl developers just to keep up with basic fixes and maintenance, none of which would go to me personally due to nonprofit regulations.

If this is something that enough people are willing to support, then please let me know much you are willing to contribute and we will see where it takes us!

@craigbarratt @haarp @LordSimal @michieltimmers @squash @Benoit75 @rgristroph @freelock @tonimore @wforumw @aqueos @futureweb @smurfix @opoplawski @apio-sys @wltechblog @moisseev @klaernie @NicolasGoeddel @nixcamic 

Thanks,
~ Will Braswell

**Comment by proserver1** [comment]
we would support this (financially)

**Comment by rgristroph** [comment]
I would put in some money towards this.

**Comment by freelock** [comment]
We could put some money towards this, in the $100-$200/year range...

**Comment by aqueos** [comment]
Aqueos can take his share in it too like 1K€/year.

**Comment by wbraswell** [comment]
@aqueos 
Thank you very much for your pledge of support!
Out of curiosity, is Aqueos the name of your company or is it simply your online username?

**Comment by futureweb** [comment]
Hi @wbraswell,  

Futureweb GmbH would also be willing to contribute $100–200 USD/y. Even though we currently have zero issues with our BackupPC setup (thanks to a fully isolated environment also safe), and we’re not really missing any features at the moment, it would definitely be nice to see the project get some new momentum instead of fading away.

Thanks for looking into this!

Best regards,
Andreas

**Comment by jgonyea** [comment]
Hi @wbraswell ,

Hart Credit Technology will also be willing to contribute $720 USD/y.

**Comment by craigbarratt** [comment]
I appreciate people offering some financial support.  However, I'm hoping that will not be necessary.

I just sent this announcement to the user and announce lists.

> BackupPC community,
> 
> As you all know, I haven't been able to commit any time to supporting BackupPC for the last several years.
> 
> I'm excited to announce that Ged Haywood @GWHAYWOOD, a long-time user and participant on the BackupPC user list, has agreed to take over supporting BackupPC.  I'll try to help him when I can.  He now has admin privileges on the three github repositories.
> 
> Hopefully some of the outstanding PRs and easy fixes will get some attention now.
> 
> Thanks to Ged for taking this on!  I appreciate everyone who has helped over the years with developing features, PRs, testing or helping users.  Both of us would be happy to hear from anyone else who can also help.
> 
> Cheers,
> Craig

**Comment by proserver1** [comment]
thank you for all your work!

**Comment by wbraswell** [comment]
Thanks for the update @craigbarratt !

Howdy @GWHAYWOOD !  We love to connect with fellow Perl programmers, would you be interested in a call just to get to know one another?

**Comment by nixcamic** [comment]
Awesome! I'm glad this will still be maintained! I work with non-profits in the developing world so we couldn't really afford to support it financially but thanks again to everyone who has worked on this project!

**Comment by GWHAYWOOD** [comment]
The project now has an active maintainer. :)
